### PR TITLE
Remove wrong lifetime from LintContext

### PR DESCRIPTION
**Pull request by lzutao**
Rustup https://github.com/rust-lang/rust/pull/61735

changelog: none

### CODE REVIEW

File: clippy_lints/src/format.rs [hunk 1]
```diff
@@ -92,7 +92,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UselessFormat {
     }
 }
 
-fn span_useless_format<'a, 'tcx: 'a, T: LintContext<'tcx>>(cx: &'a T, span: Span, help: &str, mut sugg: String) {
+fn span_useless_format<T: LintContext>(cx: &T, span: Span, help: &str, mut sugg: String) {
```
Comment (eddyb) [inline clippy_lints/src/format.rs]: cc @oli-obk `T` should be named `C` IMO, or maybe `cx: &impl LintContext` can be used (and, well, `LintContext` should probably be just `LintCx`).

**Comment by phansch** [comment]
Looks like a new breakage introduced by rustc?

**Comment by eddyb** [comment]
@phansch Yeah I'd recommend rustup-ing to just before rust-lang/rust#61735 and *then* apply the changes here, although I'm surprised clippy didn't appear as broken before rust-lang/rust#61735 was merged.

**Comment by lzutao** [comment]
Closed in favor of #4198 .

**Pull request closed without merging**